pkg/apis/sedna/v1alpha1/objectsearchservice_types.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2021 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:shortName=os
+// +kubebuilder:subresource:status
+
+// ObjectSearchService describes the data that a objectsearchservice resource should have
+type ObjectSearchService struct {
+	metav1.TypeMeta `json:",inline"`
+
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   ObjectSearchServiceSpec   `json:"spec"`
+	Status ObjectSearchServiceStatus `json:"status,omitempty"`
+}
+
+// ObjectSearchServiceSpec is a description of a objectsearchservice
+type ObjectSearchServiceSpec struct {
+	UserWorker  UserWorker  `json:"userWorker"`
+	TrackingWorkers []TrackingWorker `json:"trackingWorkers"`
+	ReidWorkers ReidWorkers `json:"reidWorkers"` 
+}
+
+// UserWorker describes the data a user worker should have
+type UserWorker struct {
+	Template v1.PodTemplateSpec `json:"template"`
+}
+
+// TrackingWorker describes the data a tracking worker should have
+type TrackingWorker struct {
+	Template v1.PodTemplateSpec `json:"template"`
+}
+
+// ReidWorkers describes the data a reid workers should have
+type ReidWorkers struct {
+	Template v1.DeploymentSpec `json:"template"`
```

Comment:
Since DeploymentSpec has template field, so I suggest to make DeploymentSpec inline which is the same comment in the example yaml.
```suggestion
	v1.DeploymentSpec `json:",inline"`
```